docs/source-pytorch/advanced/training_tricks.rst

```diff
@@ -326,7 +326,7 @@ The :class:`~pytorch_lightning.core.datamodule.LightningDataModule` class provid
         def prepare_data(self):
             MNIST(self.data_dir, download=True)
 
-        def setup(self, stage: Optional[str] = None):
+        def setup(self, stage: str):
```

Comment:
from platform import python_branch


I still see some instances that might need an update

```python
src/lightning_app/cli/pl-app-template/core/state.py:45:    stage: Optional[str] = None
src/pytorch_lightning/trainer/connectors/data_connector.py:245:        self, dataloader: Any, shuffle: Optional[bool] = None, mode: Optional[RunningStage] = None
src/pytorch_lightning/trainer/connectors/data_connector.py:296:        self, dataloader: DataLoader, shuffle: bool, mode: Optional[RunningStage] = None
src/pytorch_lightning/trainer/connectors/data_connector.py:333:        mode: Optional[RunningStage] = None,
src/pytorch_lightning/trainer/data_loading.py:30:    def prepare_dataloader(self, dataloader: Any, shuffle: bool, mode: Optional[RunningStage] = None) -> Any:
src/pytorch_lightning/profilers/profiler.py:42:        self._stage: Optional[str] = None
src/pytorch_lightning/profilers/profiler.py:151:    def setup(self, stage: str, local_rank: Optional[int] = None, log_dir: Optional[str] = None) -> None:
src/pytorch_lightning/profilers/profiler.py:157:    def teardown(self, stage: Optional[str]) -> None:
src/pytorch_lightning/profilers/advanced.py:81:    def teardown(self, stage: Optional[str]) -> None:
src/pytorch_lightning/utilities/data.py:197:    dataloader: DataLoader, sampler: Union[Sampler, Iterable], mode: Optional[RunningStage] = None
src/pytorch_lightning/utilities/data.py:207:    mode: Optional[RunningStage] = None,
src/pytorch_lightning/utilities/data.py:307:    mode: Optional[RunningStage] = None,
src/pytorch_lightning/strategies/ipu.py:241:        self, dataloader: DataLoader, sampler: Union[Sampler, Iterable], mode: Optional[RunningStage] = None
```